### PR TITLE
Feature/164768116 combine people and chat

### PR DESCRIPTION
**Pull request by vyvil**
This PR includes:
1. Merging Contact and Chat screens.
2. Updates on BottomSheet component:
2.1. Adding support for inverted content (aligned to the bottom of the sheet) by animating height.
2.2. Using native driver for animations (changed animated property from top position to transformY.
2.3. Optimised animations by reducing state updates on BottomSheet component.

Suggests on how to make animations smoothers - more than welcome :)

### CODE REVIEW

File: src/actions/notificationsActions.js [hunk 1]
```diff
@@ -176,9 +175,10 @@ export const startListeningNotificationsAction = () => {
         const { params: navParams = null } = getNavigationPathAndParamsState() || {};
         if (!navParams) return;
         dispatch({ type: SET_UNREAD_CHAT_NOTIFICATIONS_STATUS, payload: true });
-        if (!!navParams.username && navParams.username === notification.navigationParams.username) {
-          const contact = contacts.find(c => c.username === navParams.username) || {};
-          dispatch(getChatByContactAction(navParams.username, contact.id, contact.profileImage));
+        if (!!navParams.contact && !!navParams.contact.username
```
Comment (jsidorenko) [inline src/actions/notificationsActions.js]: Let's start using `lodash` `get` function to safely extract the data.
Then this code would look like: `if (get(navParams, ['contact', 'username'], '') === notification.navigationParams.username ... )`
or `get(navParams, 'contact.username', '')`.
You can even extract this into some var in order to re-use it

what do you think?

File: src/reducers/chatReducer.js [hunk 2]
```diff
@@ -176,6 +147,27 @@ export default function chatReducer(
             },
             isDecrypting: false,
             isFetching: false,
+            chats: state.data.chats
+              .map(_chat => {
+                let { lastMessage, unread } = _chat;
+                const { username: contactUsername } = _chat;
+                if (contactUsername === action.payload.username
+                  && Object.keys(state.data.messages).length
```
Comment (jsidorenko) [inline src/reducers/chatReducer.js]: `Lodash get()` could simplify this code

File: src/actions/notificationsActions.js (hunk 1 above)
Comment (vyvil) [reply src/actions/notificationsActions.js]: Done :)

File: src/components/BottomSheet/BottomSheet.js [hunk 3]
```diff
@@ -23,182 +23,307 @@ import {
...
-    const { isSheetOpen } = this.state;
-    const { onSheetOpen, onSheetClose } = this.props;
-    if (prevState.isSheetOpen !== isSheetOpen) {
-      if (isSheetOpen && onSheetOpen) {
-        onSheetOpen();
-      } else if (onSheetClose) {
-        onSheetClose();
-      }
+  componentDidUpdate(prevProps: Props) {
+    if (this.props.forceOpen && !prevProps.forceOpen) {
```
[the remark this answers is not in the document]
Comment (vyvil) [reply src/components/BottomSheet/BottomSheet.js]: Yep - updated :)

File: src/reducers/chatReducer.js (hunk 2 above)
Comment (vyvil) [reply src/reducers/chatReducer.js]: Updated this part as well :)

File: src/components/Tabs/Tabs.js [hunk 4]
```diff
@@ -119,26 +115,45 @@ const TextWrapper = styled.View`
...
   height: 24px;
   width: 24px;
   border-radius: 12px;
   background-color: ${baseColors.electricBlue};
   position: absolute;
-  top: ${Platform.select({
-    ios: '4px',
-    android: '4.5px',
-  })};
+  top 4px;
```
Comment (jsidorenko) [inline src/components/Tabs/Tabs.js]: missed the colon after `top`

Comment (vyvil) [reply src/components/Tabs/Tabs.js]: fixed :)

**Pull request merged**